.github/workflows/go-build-and-release.yml

```diff
@@ -0,0 +1,31 @@
+name: Build and Release a Version
+on:
+  push:
+    branches:
+      - master
+
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Go
+        uses: actions/setup-go@v3.0.0
+        with:
+          go-version: 1.17
+        id: go
+
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v3.0.2
```

Comment:
we don't need this setup go step - We're not doing a go build, and the semantic release is a javascript thing